base/linalg/triangular.jl

```diff
@@ -1662,35 +1662,195 @@ At_ldiv_B(::Union{UnitUpperTriangular,UnitLowerTriangular}, ::RowVector) = throw
+#     a matrix and their Fréchet derivatives", SIAM. J. Matrix Anal. & Appl.,
+#     34(3), (2013) 1341–1360.
+function powm(A0::UpperTriangular{T}, p::Real) where T<:BlasFloat
+
+    if abs(p) >= 1
+        ArgumentError("p must be a real number in (-1,1), got $p")
+    end
+
+    theta = [1.53e-5, 2.25e-3, 1.92e-2, 6.08e-2, 1.25e-1, 2.03e-1, 2.84e-1]
+    n = checksquare(A0)
+
+    A, m, s = invsquaring(A0,theta)
+    A = I - A
+
+    # Compute accurate diagonal of I - T
+    sqrt_diag!(A0,A,s)
+    for i = 1:n
+        A[i,i] = -A[i,i]
+    end
+    # Compute the Padé approximant
+    c = 0.5 * (p - m) / (2 * m - 1)
+    triu!(A)
+    S = c * A
+    Stmp = similar(S)
+    for j = m-1:-1:1
+        j4 = 4 * j
+        c = (-p - j) / (j4 + 2)
+        for i = 1:n
+            @inbounds S[i,i] = S[i,i] + 1
+        end
+        copy!(Stmp,S)
+        scale!(S,A,c)
+        A_ldiv_B!(Stmp,S.data)
+
+        c = (p - j) / (j4 - 2)
+        for i = 1:n
+            @inbounds S[i,i] = S[i,i] + 1
+        end
+        copy!(Stmp,S)
+        scale!(S,A,c)
+        A_ldiv_B!(Stmp,S.data)
+    end
+    for i = 1:n
+        S[i,i] = S[i,i] + 1
+    end
+    copy!(Stmp,S)
+    scale!(S,A,-p)
+    A_ldiv_B!(Stmp,S.data)
+    for i = 1:n
+        @inbounds S[i,i] = S[i,i] + 1
+    end
+
+    blockpower!(A0,S,p/(2^s))
+    for m = 1:s
+        A_mul_B!(Stmp.data,S,S)
+        copy!(S,Stmp)
+        blockpower!(A0,S,p/(2^(s-m)))
+    end
+    return S
+end
+^(A::LowerTriangular, p::Integer) = ^(A.', p::Integer).'
+powm(A::LowerTriangular, p::Real) = powm(A.', p::Real).'
 
 # Complex matrix logarithm for the upper triangular factor, see:
 #   Al-Mohy and Higham, "Improved inverse  scaling and squaring algorithms for
-#     the matrix logarithm", SIAM J. Sci. Comput., 34(4), (2012), pp. C153-C169.
+#     the matrix logarithm", SIAM J. Sci. Comput., 34(4), (2012), pp. C153–C169.
 #   Al-Mohy, Higham and Relton, "Computing the Frechet derivative of the matrix
-#     logarithm and estimating the condition number", SIAM J. Sci. Comput., 35(4),
-#     (2013), C394-C410.
+#     logarithm and estimating the condition number", SIAM J. Sci. Comput.,
+#     35(4), (2013), C394–C410.
 #
 # Based on the code available at http://eprints.ma.man.ac.uk/1851/02/logm.zip,
 # Copyright (c) 2011, Awad H. Al-Mohy and Nicholas J. Higham
 # Julia version relicensed with permission from original authors
 function logm{T<:Union{Float64,Complex{Float64}}}(A0::UpperTriangular{T})
-    maxsqrt = 100
     theta = [1.586970738772063e-005,
          2.313807884242979e-003,
          1.938179313533253e-002,
          6.209171588994762e-002,
          1.276404810806775e-001,
          2.060962623452836e-001,
          2.879093714241194e-001]
+
+    n = checksquare(A0)
+
+    A, m, s = invsquaring(A0,theta)
+
+    # Compute accurate diagonal of A - I
+    sqrt_diag!(A0, A, s)
+
+    # Compute the Gauss-Legendre quadrature formula
+    x, w = Base.QuadGK.gauss(Float64, m)
```

Comment:
Since it looks like `m ≤ length(theta)`, we can just precompute these and store them in a `const` array.